### PR TITLE
feat(ai-platform): add an AI platform sample

### PR DESCRIPTION
**Pull request by Symmetries**
## Description

Fixes #<ISSUE-NUMBER>

Note: It's a good idea to open an issue first for discussion.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: data-science-onramp/ai-platform/setup.py
Comment (leahecole) [inline data-science-onramp/ai-platform/setup.py:1]: What is this file? Is it discussed in the tutorial? If not, should it be included in the readme?

File: data-science-onramp/ai-platform/run.sh
Comment (bradmiro) [inline data-science-onramp/ai-platform/run.sh:5]: Maybe just an empty line before gcloud command?

File: data-science-onramp/ai-platform/run.sh
Comment (bradmiro) [inline data-science-onramp/ai-platform/run.sh:9]: Empty line between gcloud commands?

File: data-science-onramp/ai-platform/run.sh
Comment (bradmiro) [inline data-science-onramp/ai-platform/run.sh:13]: empty line before deactivate?

File: data-science-onramp/ai-platform/test_models.py
Comment (leahecole) [inline data-science-onramp/ai-platform/test_models.py:1]: this needs to be renamed `models_test.py`

File: data-science-onramp/ai-platform/trainer/sklearn_model/hptuning_config.yaml
Comment (leahecole) [inline data-science-onramp/ai-platform/trainer/sklearn_model/hptuning_config.yaml:1]: lower priority, but comments about why these are set to what they are might be helpful

**Review by bradmiro — DISMISSED** [review]

**Comment by bradmiro** [comment]
Added a couple comments but otherwise LGTM

File: data-science-onramp/ai-platform/test_models.py
Comment (leahecole) [reply data-science-onramp/ai-platform/test_models.py:1]: Actually, this should likely be split up into two files, one in each model directory with the related tests

File: data-science-onramp/ai-platform/trainer/tfkeras_model/hptuning_config.yaml
Comment (leahecole) [inline data-science-onramp/ai-platform/trainer/tfkeras_model/hptuning_config.yaml:1]: same comments re other yaml

File: data-science-onramp/ai-platform/trainer/util.py
Comment (leahecole) [inline data-science-onramp/ai-platform/trainer/util.py:9]: what do folks use this file for?

**Comment by tmatsuo** [comment]
FYI, I changed the PR title without knowing much context here, but feel free to edit it.

**Comment by bradmiro** [comment]
@leahecole think this is ready for another review. No tests yet as I figure out the best way to do so

File: data-science-onramp/ai-platform/requirements-test.txt
Comment (leahecole) [inline data-science-onramp/ai-platform/requirements-test.txt:1]: pytest has since been updated and will need to be on the latest before we merge

File: data-science-onramp/ai-platform/requirements.txt
Comment (leahecole) [inline data-science-onramp/ai-platform/requirements.txt:5]: in the runtime version list it looks like they use oauth2client 4.1.3? Should we be doing that? If we can pin, that would be preferred, because renovate bot really hates unpinned dependencies

File: data-science-onramp/ai-platform/requirements.txt
Comment (leahecole) [inline data-science-onramp/ai-platform/requirements.txt:3]: runtime version also looks like it uses 1.9.3 here

**Review by leahecole — CHANGES_REQUESTED** [review]
just nits about dependencies for now - I know there were tests before, and you said you're figuring them out - we def want to test something, but we can sit down and talk about exactly what that should look like if you want

I did not examine the code super closely rn - I'll take a second look after lunch

**Comment by busunkim96** [comment]
@Symmetries Is this still being actively worked on?

**Comment by leahecole** [comment]
Hey @busunkim96 yep Brad and I are actively working on this in the background :)

**Comment by leahecole** [comment]
No need to worry about the py-3.9 build rn it's not configured properly yet

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/task.py
Comment (leahecole) [inline data-science-onramp/ai-platform/modules/trainer/tfkeras_model/task.py:144]: Change to f string

**Review by leahecole — CHANGES_REQUESTED** [review]
ai-platform/train.csv be included, or was that done by mistake? If we are including some data to be used for a test, we may want to put it in a `testing_data` subfolder to mimic what we did for the other parts of this tutorial, and we should make it smaller.

File: data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py
Comment (leahecole) [inline data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py:37]: same comment as tfkeras regarding arg parser - is this a requirement for AI Platform/scikit learn or can it be done w/o arg parsing?

File: data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py
Comment (leahecole) [inline data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py:58]: type hints if we can

File: data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py
Comment (leahecole) [inline data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py:63]: same comments as tfkeras wrt x and y

**Review by leahecole — CHANGES_REQUESTED** [review]

File: data-science-onramp/ai-platform/modules/trainer/utils.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/modules/trainer/utils.py:19]: I've seen 'x' and 'y' quite often used for features and target respectively. Combined with the comment about it, it might be enough explanation.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/sklearn_test.py:47]: Instead of the discovery library, this should migrated to use uCAIP (and later Model Builder SDK) at some point.

Comment (ivanmkc) [reply data-science-onramp/ai-platform/sklearn_test.py:47]: Let me know if you need help on this.

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/tfkeras_test.py:47]: ditto on migrating to uCAIP

**Comment by ivanmkc** [comment]
It looks like `data-science-onramp/ai-platform/train.csv` is being checked into the repo. 

Is it good practice to check in non-code files like this into the repo? I typically see it uploaded to GCS and downloaded in the script. Not sure what the guidelines are but seems fishy.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py:47]: There's a lot of other components of this tutorial, such as composer operators, that don't currently work with uCAIP. We want to launch this as-is and hold off from migrating the whole thing to uCAIP for now.

File: data-science-onramp/ai-platform/modules/trainer/utils.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/utils.py:12]: Ack, removed.

File: data-science-onramp/ai-platform/modules/trainer/utils.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/utils.py:19]: Ack

File: data-science-onramp/ai-platform/modules/trainer/utils.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/utils.py:26]: Ack

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py:44]: Ack

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py:83]: @ivanmkc what do you think?

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/task.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/tfkeras_model/task.py:30]: Yep! The product provides its own [command line args](https://cloud.google.com/ai-platform/training/docs/packaging-trainer). This is only relevant within the service itself, not in the testing environment.

File: data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py:37]: Ack above

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/sklearn_test.py:48]: Make sense!

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/tfkeras_test.py]: Done!

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py]: Update: migrated the sample!

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/tfkeras_test.py:8]: Remove this if unneeded

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/sklearn_test.py:9]: Remove this if unneeded

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/sklearn_test.py:85]: LGTM

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/sklearn_test.py:90]: In the MBSDK, we have a synchronous method which should remove the need for things like this.

**Comment by ivanmkc** [comment]
AIP calls LGTM.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/sklearn_test.py]: Thanks!

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/tfkeras_test.py:47]: Thanks! Great stuff.

I think we're just missing code to cleanup the artifacts when done.
- Remove the training job
- Remove the trained model

I think the uCAIP notebook I sent you should have example code for cleaning those up.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/sklearn_test.py:48]: @bradmiro I think this might still be needed to delete the output model right?

Also, please delete the custom training pipeline here or the test will keep making more of them.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/sklearn_test.py:92]: Also, please delete the custom training pipeline here or the test will keep making more of them.

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (ivanmkc) [inline data-science-onramp/ai-platform/tfkeras_test.py:90]: Please delete the custom training pipeline here or the test will keep making more of them.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/sklearn_test.py:92]: reference: https://github.com/googleapis/python-aiplatform/blob/master/samples/snippets/delete_custom_job_sample.py

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/tfkeras_test.py]: Done!

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py:9]: Done!

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py]: Ack

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py]: Fixed!

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py]: Done!

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/tfkeras_test.py]: Done!

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py]: Sorry just seeing this. I don't know if this tutorial's scope covers the internals of the custom model. If not, then it might be okay to just note that this is just one example of a model definition, written using the Keras API.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/sklearn_test.py]: Looks good!

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/sklearn_test.py]: LGTM

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (ivanmkc) [reply data-science-onramp/ai-platform/tfkeras_test.py]: Nice!

**Review by ivanmkc — APPROVED** [review]
LGTM

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (leahecole) [inline data-science-onramp/ai-platform/sklearn_test.py:73]: you either need to add the force true flag or wrap this in a try/except for a NotFound

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (leahecole) [inline data-science-onramp/ai-platform/sklearn_test.py:114]: Increase timeout to 1 min and also consider adding a comment that this may not be best practice and is subject to change as the service evolves

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (leahecole) [inline data-science-onramp/ai-platform/tfkeras_test.py:113]: Increase timeout to 1 min and also consider adding a comment that this may not be best practice and is subject to change as the service evolves

**Review by leahecole — CHANGES_REQUESTED** [review]
@ivanmkc @bradmiro is using the arg parser/command line args a requirement for uCAIP/CAIP? We generally try to avoid having samples with command line args because they're harder to maintain.

File: data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/sklearn_model/task.py:60]: Splitting up or leaving in one function are both appropriate. As for failure, the tests will still properly clean up no matter where in the process this fails.

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py:57]: Explained in the documentation.

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py:63]: Explained in more detail in the documentation.

File: data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py:100]: We can clarify a bit there.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py:73]: Ack.

File: data-science-onramp/ai-platform/sklearn_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/sklearn_test.py:114]: Ack.

File: data-science-onramp/ai-platform/tfkeras_test.py
Comment (bradmiro) [reply data-science-onramp/ai-platform/tfkeras_test.py:113]: Ack.

**Comment by bradmiro** [comment]
> @ivanmkc @bradmiro is using the arg parser/command line args a requirement for uCAIP/CAIP? We generally try to avoid having samples with command line args because they're harder to maintain.

uCAIP relies on some degree of command line args for directory locations.

**Review by leahecole — APPROVED** [review]

**Pull request merged**
